test/plugins/cxxrandom.lua

```diff
@@ -0,0 +1,65 @@
+local rng = require('plugins.cxxrandom')
+
+function test.cxxrandom_distributions()
+    rng.normal_distribution:new(0,5)
+    rng.real_distribution:new(-1,1)
+    rng.int_distribution:new(-20,20)
+    rng.bool_distribution:new(0.00000000001)
+    rng.num_sequence:new(-1000,1000)
+    -- no errors, no problem
+end
+
+function test.cxxrandom_seed()
+    local nd = rng.normal_distribution:new(0,500000)
+    local e1 = rng.MakeNewEngine(1)
+    local e2 = rng.MakeNewEngine(1)
+    local e3 = rng.MakeNewEngine(2)
+    local g1 = rng.crng:new(e1, true, nd)
+    local g2 = rng.crng:new(e2, true, nd)
+    local g3 = rng.crng:new(e3, true, nd)
+    expect.eq(g1:next(), g2:next())
+    expect.ne(g1:next(), g3:next())
+end
+
+function test.cxxrandom_ranges()
+    local e1 = rng.MakeNewEngine(1)
+    local g1 = rng.crng:new(e1, true, rng.normal_distribution:new(0,1))
+    local g2 = rng.crng:new(e1, true, rng.real_distribution:new(-5,5))
+    local g3 = rng.crng:new(e1, true, rng.int_distribution:new(-5,5))
+    local g4 = rng.crng:new(e1, true, rng.num_sequence:new(-5,5))
+    for i = 1, 10000 do
+        local a = g1:next()
+        local b = g2:next()
+        local c = g3:next()
+        local d = g4:next()
+        expect.ge(a, -5)
+        expect.ge(b, -5)
+        expect.ge(c, -5)
+        expect.ge(d, -5)
+        expect.le(a, 5)
+        expect.le(b, 5)
+        expect.le(c, 5)
+        expect.le(d, 5)
+    end
+    local gb = rng.crng:new(e1, true, rng.bool_distribution:new(0.00000000001))
+    for i = 1, 10 do
+        expect.false_(gb:next())
```

Comment:
We can test that the answer is either == true or == false (i.e. that it is a boolean), but we cannot guarantee the exact value.

Reply:
Seeded